src/app.js

```diff
@@ -6,8 +26,356 @@ import _ from 'underscore';
+    },
+
+    error: (model, response) => {
+      console.log('Failed to save reservation! Server response:');
+      $('#reservation-status').removeClass('success');
+      $('#reservation-status').addClass('failure');
+
+      handleValidationFailures(response.responseJSON["errors"]);
+
+    },
+  });
+};
+
+const readReservationData = function readReservationData() {
+  const reservationData = {};
+  RESERVATION_FIELDS.forEach((field) => {
+    console.log(field);
+    const inputElement = $(`#add-reservation-form input[name="${ field }"]`);
+    const value = inputElement.val();
+    console.log(value);
+    if (value != '') {
+      reservationData[field] = value;
+    }
+    // making the tripId equal an empty string
+    // inputElement.val('');
+  });
+  console.log(reservationData);
+  return reservationData;
+}
+
+const handleReservationValidationFailures = function handleValidationFailures(errors) {
+  // Since these errors come from a Rails server, the strucutre of our
+  // error handling looks very similar to what we did in Rails.
+
+  console.log('in handlereservationvalidation');
+  for (let field in errors) {
+    for (let problem of errors[field]) {
+      reportReservationStatus('error', `${field}: ${problem}`);
+    }
+  }
+};
+
+const reportReservationStatus = function reportReservationStatus(status, message) {
+  console.log(`Reporting ${ status } status: ${ message }`);
+  // Should probably use an Underscore template here.
+  const statusHTML = `<p class="${ status }">${ message }</p>`;
+  console.log(statusHTML);
+  // note the symetry with clearStatus()
+  // $('#reservation-status').html('');
+
+  $('#reservation-status').append(statusHTML);
+  $('#reservation-status').removeClass('success')
+
+  $('#reservation-status').addClass('failure')
+  $('#reservation-status').show();
+};
+
+
+const addTripHandler = function(event) {
+  event.preventDefault();
+  // $('#myModal').removeClass('block')
+  // $('#add-trip-form').removeClass('none')
+  $('#status-messages').html('')
+  console.log('in addtriphandler');
+
+  const trip = new Trip(readFormData());
+  console.log(trip);
+
+  if (!trip.isValid()) {
+    handleValidationFailures(trip.validationError);
+    return;
+  }
+  console.log('about to add!');
+  console.log(trip);
+
+  //in response to API validations, not to model validations
+  trip.save({}, {
+    success: (model, response) => {
+      console.log('Successfully saved trip!');
+      console.log(trip);
+      tripList.add(trip);
+      $('#status-messages').addClass('success')
+      reportStatus('success', 'Successfully saved trip!');
+    },
+    error: (model, response) => {
+      console.log('Failed to save trip! Server response:');
+      //NEED TO FIND A WAY TO
+      $('#status-messages').removeClass('success')
+      $('#status-messages').addClass('failure')
+      $('#status-messages').append(`Failed to save trip`)
+      console.log(response);
+      tripList.remove(model)
```

Comment:
Just a note, you can also do `model.destroy()` to remove the model from the collection and remove any listeners.